Foundation/NSKeyedArchiver.swift

```diff
@@ -487,7 +487,7 @@ open class NSKeyedArchiver : NSCoder {
         let className = NSStringFromClass(clsv)
         let mappedClassName = _classNameForClass(clsv)
         
-        if mappedClassName != nil && mappedClassName != className {
+        if let mappedClassName = mappedClassName && mappedClassName != className {
```

Comment:
`&&` is not valid in this case; it should be `if let mappedClassName = mappedClassName, mappedClassName != className {`